superset/views/core.py

```diff
@@ -1109,29 +1116,27 @@ def get_samples(self, viz_obj):
             'data': viz_obj.get_samples(),
         })
 
+    def check_datasource_permission(self, datasource, user=None):
+        if not security_manager.datasource_access(datasource, user):
+            return json_error_response(
```

Comment:
Shouldn't this raise instead of returning?

Reply:
I was returning an error json previously. I can change this to a `raise` for the new `handle_superset_exception` decorator to catch and handle.